src/main/java/tools/map/xml/creator/MapXmlHelper.java

```diff
@@ -456,438 +462,328 @@ static void clearGameSettings() {
-      final Node childNode = children.item(i);
-      final String childNodeName = childNode.getNodeName();
-      if (childNodeName.equals(XML_NODE_NAME_INFO)) {
-        final HashMap<String, String> infoAttr = getAttributesMap(childNode.getAttributes());
-        for (final Entry<String, String> infoAttrEntry : infoAttr.entrySet()) {
-          getXmlStringsMap().put("info_@" + infoAttrEntry.getKey(), infoAttrEntry.getValue());
-        }
-      } else if (childNodeName.equals(XML_NODE_NAME_RESOURCE_LIST)) {
-        final NodeList resourceNodes = childNode.getChildNodes();
-        for (int j = 0; j < resourceNodes.getLength(); ++j) {
-          final Node resourceNode = resourceNodes.item(j);
-          if (resourceNode.getNodeName().equals(XML_NODE_NAME_RESOURCE)) {
-            getResourceList().add(resourceNode.getAttributes().item(0).getNodeValue());
-          }
-        }
-      } else if (childNodeName.equals(XML_NODE_NAME_MAP)) {
-        parseMapNode(childNode.getChildNodes());
-        stepToGo = MapXmlCreator.getMaxGameStep(stepToGo, (getTerritoryConnectionsMap().isEmpty()
-            ? GameStep.TERRITORY_DEFINITIONS : GameStep.TERRITORY_CONNECTIONS));
-      } else if (childNodeName.equals(XML_NODE_NAME_PLAYER_LIST)) {
-        parsePlayerListNode(childNode.getChildNodes());
-        stepToGo = MapXmlCreator.getMaxGameStep(stepToGo,
-            GameStep.PLAYERS_AND_ALLIANCES);
-      } else if (childNodeName.equals(XML_NODE_NAME_PRODUCTION)) {
-        putNodesToProductionFrontiers(childNode.getChildNodes());
-        stepToGo = MapXmlCreator.getMaxGameStep(stepToGo,
-            (getProductionFrontiersMap().isEmpty() ? GameStep.UNIT_DEFINITIONS : GameStep.PRODUCTION_FRONTIERS));
-      } else if (childNodeName.equals(XML_NODE_NAME_GAME_PLAY)) {
-        putNodesToPlayerSequence(childNode.getChildNodes());
-        stepToGo = MapXmlCreator.getMaxGameStep(stepToGo,
-            (getPlayerSequenceMap().isEmpty() ? GameStep.UNIT_ATTACHMENTS : GameStep.TERRITORY_PRODUCTION));
-      } else if (childNodeName.equals(XML_NODE_NAME_ATTACHMENT_LIST)) {
-        final NodeList attachmentListChildNodes = childNode.getChildNodes();
-        for (int j = 0; j < attachmentListChildNodes.getLength(); ++j) {
-          final Node attachment = attachmentListChildNodes.item(j);
-          if (attachment.getNodeName().equals(XML_NODE_NAME_ATTACHMENT)) {
-            parseAttachmentNode(attachment);
-          }
+      private String playerName = null;
+      private final List<String> frontierRules = new ArrayList<>();
+
+
+      private CanalTerritoriesTuple canalDef = null;
+      private String newCanalName = null;
+      private SortedSet<String> newLandTerritories = new TreeSet<>();
+      private String attachmentAttachTo = null;
+
+      private List<String> settingValues = new ArrayList<>();
+
+      @Override
+      protected void handleInfo(Attributes attributes) throws GameParseException {
```

Comment:
Sorry if I wasn't clear about this...  I'm not suggesting to remove the exception specification from the **base class** methods for the reasons you specified.  I'm simply saying that the **subclass** methods do not throw a `GameParseException`, and thus they don't need to include the exception specification.  Since the subclass is anonymous, no other class can override these methods, and the "generic" exception specifications don't need to be present.

Not a big deal.  It's just convenient to be able to look at the signature of a concrete method and tell at a glance what checked exceptions it throws.  In this case, one has to examine the code before coming to the conclusion that, in fact, no checked exceptions are thrown.